http-api/src/v1/sessions.rs

```diff
@@ -0,0 +1,171 @@
+use axum::{Extension, Json, Router};
+use chrono::{DateTime, Utc};
+use ethers_core::utils::hex;
+use hyper::http::uri::Authority;
+use serde_json::json;
+use siwe::Message;
+
+use crate::auth::{AuthRequest, AuthState, Session};
+use crate::axum_extra::Path;
+use crate::{Context, Error};
+
+pub const UNAUTHORIZED_SESSIONS_EXPIRATION: Duration = Duration::from_secs(60);
+
+/*
+ *
+fn session_filters(ctx: Context) -> BoxedFilter<(impl Reply,)> {
+    session_create_filter(ctx.clone())
+        .or(session_signin_filter(ctx.clone()))
+        .or(session_get_filter(ctx))
+        .boxed()
+}
+
+/// `POST /sessions`
+fn session_create_filter(
+    ctx: Context,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    warp::post()
+        .map(move || ctx.clone())
+        .and(path::end())
+        .and_then(session_create_handler)
+}
+
+/// `PUT /sessions/:session-id`
+fn session_signin_filter(
+    ctx: Context,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    warp::put()
+        .map(move || ctx.clone())
+        .and(path::param::<String>())
+        .and(path::end())
+        .and(warp::body::json())
+        .and_then(session_signin_handler)
+}
+
+/// `GET /sessions/:session-id`
+fn session_get_filter(
+    ctx: Context,
+) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
+    warp::get()
+        .map(move || ctx.clone())
+        .and(path::param::<String>())
+        .and(path::end())
+        .and_then(session_get_handler)
+}
+*/
+
+pub fn router(ctx: Context) -> Router {
+    Router::new()
+        .route("/sessions", post(session_create_handler))
+        .route(
+            "/sessions/:id",
+            get(session_get_handler).put(session_signin_handler),
+        )
+        .layer(Extension(ctx))
+}
+
+/// Create session.
+/// `POST /sessions`
+async fn session_create_handler(Extension(ctx): Extension<Context>) -> impl IntoResponse {
+    let expiration_time =
+        Utc::now() + chrono::Duration::from_std(UNAUTHORIZED_SESSIONS_EXPIRATION).unwrap();
+    let mut sessions = ctx.sessions.write().await;
+    let (session_id, nonce) = create_session(&mut sessions, expiration_time);
+
+    Json(json!({ "id": session_id, "nonce": nonce }))
+}
+
+/// Get session.
+/// `GET /sessions/:session_id`
+async fn session_get_handler(
+    Extension(ctx): Extension<Context>,
+    Path(id): Path<String>,
+) -> impl IntoResponse {
+    let sessions = ctx.sessions.read().await;
+    let session = sessions.get(&id).ok_or(Error::NotFound)?;
+
+    match session {
+        AuthState::Authorized(session) => {
+            Ok::<_, Error>(Json(json!({ "id": id, "session": session })))
+        }
+        AuthState::Unauthorized {
+            nonce,
+            expiration_time,
+        } => Ok::<_, Error>(Json(
+            json!({ "id": id, "nonce": nonce, "expirationTime": expiration_time }),
+        )),
+    }
+}
+
+/// Update session.
+/// `PUT /sessions/:session_id`
```

Comment:
Don't forget to change the `:session_id`in the docstrings as well

Reply:
https://github.com/radicle-dev/radicle-client-services/pull/168/commits/a3779ffffb7f1e3d6044a5ac0628bd359def3b69